src/EditorFeatures/Core/Intents/IIntentProcessor.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Features.Intents;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Intents
+{
+    internal interface IIntentProcessor
+    {
+        /// <summary>
+        /// For an input intent, computes the edits required to apply that intent and returns them.
+        /// </summary>
+        /// <param name="intentRequestContext">the intents with the context in which the intent was found.</param>
+        /// <returns>the edits that should be applied to the current snapshot.</returns>
+        Task<IntentResult?> ComputeEditsAsync(IntentRequestContext intentRequestContext, CancellationToken cancellationToken);
```

Comment:
@intellicode
2.  Would you like us to support handling multiple input intents (a list of `IntentRequestContext`) in a single request?  Also to support the same intent, but with different intent data.  This could theoretically be handled instead by just calling the API multiple times.  Wanted to get your thoughts on which you prefer.
3. I think we will need to support returning many results for a single input intent.  For example, in generate constructor when there is an existing ctor we offer two actions - generate a ctor which assigns all fields, or generate a ctor which delegates to the existing ctor.  Right now I'm just taking the first arbitrarily, but it may be more appropriate to display both options.  Would you be OK with us returning multiple IntentResults for each intent?

Reply:
2. I would prefer calling multiple times

3. Yes that would be fine, we will figure out a way to get these presented. If you have something on the likelyhood, we can sort them by that field, or add some characteristics in the properties, we can log them in telemetry and find out which are more likely to be taken offline and put that back in the final product.

Reply:
@pgroene 
3.  I believe we should have the order that they would have appeared in the lightbulb.  I can give them to you in that same order.

Reply:
We would be able to have multiple suggestions for one refactoring in the completion list and show the top one in the lightbulb

For the next version I would like to callback from the lightbulb and open the roslyn refactoring window on the refactoring